### ISSUE TITLE
Fails to link the app, 

### ISSUE THREAD

**Issue by rubelux** [transistorsoft/react-native-background-geolocation#328]
## Your Environment
* Plugin version: "react-native-background-geolocation": "^2.9.4",
* Platform: iOS
* OS version: 10
* Device manufacturer / model:
* React Native version (`react-native -v`): 0.43.3
* Plugin config

## Expected Behavior
It links as normal

## Actual Behavior
It fails to link, I followed in detail the installation steps

## Steps to Reproduce

1.
2.
3.
4.

## Debug logs
/<my app directory>/node_modules/react-native-background-geolocation/scripts/postlink.js:102
    project.addTargetAttribute('SystemCapabilities', systemCapabilities);
            ^
TypeError: project.addTargetAttribute is not a function

Huge thanks for any help, I really need to get it running

**Comment by christocracy** [comment]
Yes, my link function is brittle.  It's seems to have trouble with XCode 9.

Follow the Manual Setup Steps.

**Comment by rubelux** [comment]
Hi @christocracy 

I found my way through, it was a problem with another dependency.

I am trying now to get some props from the heartbeat callback bad it seems that the event is running in a totally different scope, I tried to bind it to my component scope (this) but the same error keeps coming.

Am I missing something here?

thanks for your help

Ruben
